addons/mail/static/src/models/autocomplete-input-view/autocomplete-input-view.js

```diff
@@ -0,0 +1,295 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one2one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'AutocompleteInputView',
+    identifyingFields: [['chatWindowOwnerAsNewMessageFormInput', 'discussAsMobileAddItemHeader', 'discussSidebarCategoryAsAddingItemInputView']],
+    recordMethods: {
+        doFocus() {
+            if (!this.component) {
+                return;
+            }
+            this.component.root.el.focus();
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onBlur(ev) {
+            this._hide();
+        },
+        onFocusin() {
+            this.update({ isFocused: true });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onKeydown(ev) {
+            if (ev.key === 'Escape') {
+                this._hide();
+            }
+        },
+        /**
+         * @private
+         */
+        _hide() {
+            if (this.discussAsMobileAddItemHeader) {
+                this.discussAsMobileAddItemHeader.clearIsAddingItem();
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputView) {
+                this.discussSidebarCategoryAsAddingItemInputView.onHideAddingItem();
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                this.messagingMenuAsMobileNewMessageInput.toggleMobileNewMessage();
+            }
+        },
```

Comment:
if all the rest of the code is following our general structure for view records, "hide" action on a view should be strictly equivalent to calling just `delete()`